File: #0594.longest-harmonious-subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        
        std::sort(nums.begin(), nums.end());
        
        map<int, int> count;
        vector<int> key;
        for (int n : nums) {
            if (count.find(n) == count.end()) {
                count[n] = 1;
                key.push_back(n);
            } else {
                count[n]++;
            }
        }
        
        int size = key.size();
        int result = 0;
        for (int i = 1; i < size; i++) {
            int k1 = key[i];
            int k2 = key[i-1];
            if (k1 - k2 == 1) {
                result = std::max(result, count[k1] + count[k2]);
            }
        }
        return result;
    }
};

File: #0594.longest-harmonious-subsequence.ts
function findLHS(nums: number[]): number {
    const numCount = new Map<number, number>();
    let maxLen = 0;
    for (const num of nums) {
        numCount.set(num, (numCount.get(num) || 0) + 1);
    }

    for (const num of numCount.keys()) {
        if (numCount.has(num + 1)) {
            maxLen = Math.max(maxLen, numCount.get(num)!
                + numCount.get(num + 1)!);
        }
    }

    return maxLen;
};
